GetAssetTrigger/index.ts

```diff
@@ -0,0 +1,74 @@
+import { AzureFunction, Context, HttpRequest } from "@azure/functions"
+import axios, { AxiosError } from "axios"
+import type { Asset } from "@equinor/data-marketplace-models"
+import { config } from "../config"
+
+const GetAssetTrigger: AzureFunction = async function (context: Context, req: HttpRequest): Promise<void> {
+  const { id } = context.bindingData
+
+  try {
+    const { data: collibraAsset } = await axios.get(`${config.COLLIBRA_BASE_URL}/assets/${id}`, {
+      headers: {
+        authorization: req.headers.authorization,
+      },
+    })
+
+    const asset: Asset = {
+      createdAt: new Date(collibraAsset.createdOn),
+      description: "",
+      excerpt: "",
+      id: collibraAsset.id,
+      provider: {
+        id: "",
+        name: "Collibra",
+      },
+      qualityScore: 0,
+      rating: 0,
+      tags: [],
+      type: {
+        id: collibraAsset.type.id,
+        name: collibraAsset.type.name,
+      },
+      updatedAt: new Date(collibraAsset.lastModifiedOn),
+      updateFrequency: "",
+    }
+
+    const { data: collibraAttrs } = await axios.get<{ results: any[] }>(`${config.COLLIBRA_BASE_URL}/attributes`, {
+      headers: {
+        authorization: req.headers.authorization,
+      },
+      params: {
+        assetId: id,
+      },
+    })
+
+    asset.description = collibraAttrs.results.find((attr) => attr.type.name === "Description")?.value ?? ""
+    asset.updateFrequency = collibraAttrs.results.find((attr) => attr.type.name === "Timeliness").value ?? ""
+
+    context.res = {
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(asset),
+    }
+  } catch (e) {
+    if (e instanceof AxiosError) {
+      context.log(`Request to ${e.config.url} failed with status code ${e.response.status}`)
+      context.log(e.response.data)
+    } else {
+      context.log(e)
+    }
+
+    context.res = {
+      status: e.response?.status ?? 500,
```

Comment:
Not sure about this one. If we get 404 from the api and return that to the client, the client could believe that the route was not found. And a 500 from the api does not mean that we have internal error.

Reply:
the only way we'd get 404 from collibra, is if the asset doesn't exist. in which case we should respond 404 too, no?

Reply:
I was thinking more in general, if we should just forward the status code from the api. Some times it will be semantically correct and some times not. I guess this is something we should discuss on a more general level though. We could include it in the "functional approach to error handling" workshop 😀

Reply:
are we good to merge this now, or would you want a change?

Reply:
Merge it now, but make a note that we should discuss this in the team.

Reply:
ok. can you set that up and invite the team so we don't keep bringing it up every time there's non-fp error handling in a pull request?

Reply:
also, if it's good to merge, can you approve, please?